artemis-core-client/src/main/java/org/apache/activemq/artemis/api/core/RefCountMessage.java

```diff
@@ -58,55 +108,35 @@ public int getDurableCount() {
    public RefCountMessage getParentRef() {
       return parentRef;
    }
-   // I am usually against keeping commented out code
-   // However this is very useful for me to debug referencing counting.
-   // Uncomment out anything between #ifdef DEBUG and #endif
 
-   // #ifdef DEBUG -- comment out anything before endif if you want to debug REFERENCE COUNTS
-   //final ConcurrentHashSet<Exception> upSet = new ConcurrentHashSet<>();
-   // #endif
+   protected void onUp() {
+      if (debugStatus != null) {
+         debugStatus.up(debugString());
+      }
+   }
 
-   private void onUp() {
-      // #ifdef DEBUG -- comment out anything before endif if you want to debug REFERENCE COUNTS
-      // upSet.add(new Exception("upEvent(" + debugString() + ")"));
-      // #endif
+   protected void onFired() {
```

Comment:
Perhaps onDelete[d]? onRelease?

Calling it onFired() seems weird since its so often called when the 'releaseComplete()' being gated by the 'fired' variable apparently _has not_ been fired.

Reply:
I am going to rename this as accountedFor...


although I'm improving this... I'm only going to report an issue if there was at least on onUp, so I will not need this as often... probably it will be called by onFired only.

Reply:
I have pushed my latest changes now.